File: Rgz2/app/src/main/java/com/example/rgz/MainActivity.java
package com.example.rgz;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

// Главный экран "AI Dev Advisor". ЛР1-2: текст, проект, счётчик инструментов в базе.
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        int count = new DBHelper(this).getAllTools().size();
        ((TextView) findViewById(R.id.toolsCount)).setText("В базе инструментов: " + count);

        findViewById(R.id.btnSelect).setOnClickListener(v ->
                startActivity(new Intent(this, ParamsActivity.class)));
        findViewById(R.id.btnCatalog).setOnClickListener(v ->
                startActivity(new Intent(this, CatalogActivity.class)));
        findViewById(R.id.btnMap).setOnClickListener(v ->
                startActivity(new Intent(this, MapsActivity.class)));
    }
}

File: lab07/Kotlin/app/src/main/java/com/example/lab07/MainActivity.kt
package com.example.lab07

import android.os.Bundle
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.PopupMenu

class MainActivity : AppCompatActivity() {

    private lateinit var imageView: ImageView
    private lateinit var textView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        imageView = findViewById(R.id.imageView)
        textView = findViewById(R.id.textView)

        imageView.setOnClickListener {
            val popupMenu = PopupMenu(this, imageView)

            popupMenu.menu.add(1, 1, 1, "slot1")
            popupMenu.menu.add(1, 2, 2, "slot2")

            popupMenu.setOnMenuItemClickListener {
                textView.text = getString(R.string.selected_item, it.title)
                true
            }

            popupMenu.show()
        }
    }
}